# Import Libaries

In [13]:
import kagglehub
import pandas as pd

from pathlib import Path

# 0. Load Data

In [9]:
path = kagglehub.dataset_download(
    "abdallahwagih/ucf101-videos",
    output_dir="data"
)

DATA_PATH = Path(path)

print(DATA_PATH)

data


# 1. Data Inspection

## Inspect Downloaded Folder and Files

In [11]:
print(list(DATA_PATH.iterdir()))

[WindowsPath('data/.complete'), WindowsPath('data/test'), WindowsPath('data/test.csv'), WindowsPath('data/train'), WindowsPath('data/train.csv')]


In [15]:
print("Train:")
print(list((DATA_PATH / "train").iterdir()))

print("\nTest:")
print(list((DATA_PATH / "test").iterdir()))

Train:
[WindowsPath('data/train/v_CricketShot_g08_c01.avi'), WindowsPath('data/train/v_CricketShot_g08_c02.avi'), WindowsPath('data/train/v_CricketShot_g08_c03.avi'), WindowsPath('data/train/v_CricketShot_g08_c04.avi'), WindowsPath('data/train/v_CricketShot_g08_c05.avi'), WindowsPath('data/train/v_CricketShot_g08_c06.avi'), WindowsPath('data/train/v_CricketShot_g08_c07.avi'), WindowsPath('data/train/v_CricketShot_g09_c01.avi'), WindowsPath('data/train/v_CricketShot_g09_c02.avi'), WindowsPath('data/train/v_CricketShot_g09_c03.avi'), WindowsPath('data/train/v_CricketShot_g09_c04.avi'), WindowsPath('data/train/v_CricketShot_g09_c05.avi'), WindowsPath('data/train/v_CricketShot_g09_c06.avi'), WindowsPath('data/train/v_CricketShot_g09_c07.avi'), WindowsPath('data/train/v_CricketShot_g10_c01.avi'), WindowsPath('data/train/v_CricketShot_g10_c02.avi'), WindowsPath('data/train/v_CricketShot_g10_c03.avi'), WindowsPath('data/train/v_CricketShot_g10_c04.avi'), WindowsPath('data/train/v_CricketShot_

In [16]:
train_df = pd.read_csv(DATA_PATH / "train.csv")
test_df = pd.read_csv(DATA_PATH / "test.csv")

print(train_df.head())

print(test_df.head())

                  video_name          tag
0  v_CricketShot_g08_c01.avi  CricketShot
1  v_CricketShot_g08_c02.avi  CricketShot
2  v_CricketShot_g08_c03.avi  CricketShot
3  v_CricketShot_g08_c04.avi  CricketShot
4  v_CricketShot_g08_c05.avi  CricketShot
                  video_name          tag
0  v_CricketShot_g01_c01.avi  CricketShot
1  v_CricketShot_g01_c02.avi  CricketShot
2  v_CricketShot_g01_c03.avi  CricketShot
3  v_CricketShot_g01_c04.avi  CricketShot
4  v_CricketShot_g01_c05.avi  CricketShot


### Info
The dataset already contains prepared **train** and **test** sets, together with `train.csv` and `test.csv`.
- The **video folders** contain the actual video files that will be loaded and processed by our model.
- The **CSV files** contain information that tells us which video belongs to which class (label). We will use them to connect each video with its correct target label during training and evaluation.

## Basic Data Inspection

In [27]:
print("="*50)
print("DATA INSPECTION")
print("="*50)

print("\nShapes:")
print("Train: ", train_df.shape)
print("Test: ", test_df.shape)

print("\nNull Values:")
print("Train: ", train_df.isnull().sum())
print("\nTest: ", test_df.isnull().sum())

print("\nNumber of classes: ")
print('Classes: ', len(train_df["tag"].unique()))
print(list(train_df["tag"].unique()))

print("\nClass distibution:")
print("Train: ", train_df["tag"].value_counts())
print("\nTest: ", test_df["tag"].value_counts())

DATA INSPECTION

Shapes:
Train:  (594, 2)
Test:  (224, 2)

Null Values:
Train:  video_name    0
tag           0
dtype: int64

Test:  video_name    0
tag           0
dtype: int64

Number of classes: 
Classes:  5
['CricketShot', 'PlayingCello', 'Punch', 'ShavingBeard', 'TennisSwing']

Class distibution:
Train:  tag
Punch           121
PlayingCello    120
CricketShot     118
ShavingBeard    118
TennisSwing     117
Name: count, dtype: int64

Test:  tag
CricketShot     49
TennisSwing     49
PlayingCello    44
ShavingBeard    43
Punch           39
Name: count, dtype: int64


### Info
The dataset consists of **818 observations**: **594 for training** and **224 for testing**. There are **no missing values** in either the training or test dataset. The dataset contains **5 classes**, which are **evenly distributed** across both the training and test sets.